pkg/api/dtos/models.go

```diff
@@ -77,7 +77,7 @@ func GetGravatarUrlWithDefault(text string, defaultText string) string {
 }
 
 func IsHiddenUser(userLogin string, signedInUser *models.SignedInUser, cfg *setting.Cfg) bool {
-	if signedInUser.IsGrafanaAdmin || userLogin == signedInUser.Login {
+	if userLogin == "" || signedInUser.IsGrafanaAdmin || userLogin == signedInUser.Login {
```

Comment:
Would be good having a test of this method. 

In addition it feels a bit fragile/awkward that this method would be called for any permission, should only be called for user permissions. WDYT?

Reply:
Yes you're right, I added the test in the dashboard permissions file. Now, I'm not sure if we should keep this line or remove it.

Reply:
I agree, we should def call this only when we need to.

Reply:
Yes, I added a check for dashboard / folder permissions in [this commit](https://github.com/grafana/grafana/pull/31132/commits/bdb6542ec620803c61e1309e0bb3f77eb941609c) and I made a PR for Enterprise too.

Reply:
> I'm not sure if we should keep this line or remove it.

If someone uses the API with an API key the `signedInUser.Login` would be empty. But doesn't really matter because permissions shouldn't have an empty `userLogin` is it's a user permissions. This extra check wouldn't hurt I would say :)